src/app/operators/components/operator/operator.component.ts

```diff
@@ -32,13 +33,23 @@ export class OperatorComponent implements OnInit {
   ) {}
 
   ngOnInit() {
+    this.operatorsMap = this.operators.reduce(
+      (acc: { string?: OperatorDoc }, value: OperatorDoc) => {
+        acc[value.name] = value;
+        return acc;
+      },
+      {}
+    );
+
     this._activatedRoute.params
       .pipe(pluck('operator'))
       .subscribe((name: string) => {
-        this.operator =
-          this.operators.filter(
-            (operator: OperatorDoc) => operator.name === name
-          )[0] || this.notfound();
+        if (name in this.operatorsMap) {
```

Comment:
@feloy - Can use `Map` instead of object to create a map of operators. This works as well, but map being explicit.